cmd/metacache-server-pool.go

```diff
@@ -189,7 +189,7 @@ func (z *erasureServerPools) listPath(ctx context.Context, o listPathOptions) (e
 					if err != nil {
 						return true
 					}
-					oFIV, err := existing.fileInfo(o.Bucket)
+					oFIV, err := other.fileInfo(o.Bucket)
```

Comment:
Note: Bugfix.

Reply:
oh @klauspost - can a test case be added for this if possible? - @vadmeste ^^ this is possibly the issue with stale prefixes?

Reply:
BTW @klauspost I need more time to test this PR for any corruption holes, just this fix alone can it be sent separately?